irodsConnector/session.py

```diff
@@ -15,28 +15,64 @@
 
 class Session(object):
     """Irods session operations """
-    _session = None
-    context = utils.context.Context()
 
-    def __init__(self, password=''):
+    def __init__(self, context: utils.context.Context, password=None, greet=False):
         """ iRODS authentication with Python client.
 
         Parameters
         ----------
         password : str
             Plain text password.
+        context : utils.context Context object containing valid
+            irods environment and ibridges configuration 
+        greet: bool
+            Print some greeting and irods info to prompt
 
-        The 'password' property can autoload from its cache, but can be
+        The 'password' is autoloaded from its cache, but can be
         overridden by `password` argument.  The iRODS authentication
         file is expected in the standard location (~/.irods/.irodsA) or
         to be specified in the local environment with the
         IRODS_AUTHENTICATION_FILE variable.
 
         """
         self._password = password
+        self._ienv = context.irods_environment.config
+        self._conf = context.ibridges_configuration.config
+        self._irods_env_file = context.irods_env_file
+        self._irods_session = None
+
+        if self._ienv == None:
+            raise ValueError("ERROR SESSION: no irods environment")
+        if self._conf == None:
+            raise ValueError("ERROR SESSION: no ibridges configuration")
```

Comment:
The check for valid configurations is done only once upon initialization here.  As I mentioned for lines 39-40, for a new session, a new instance of this class will be required or separate methods will be needed to reset and check the configurations each time the session is closed.  `@property`s are better suited for this as they have on-demand access to the current state of the **context**.